### PR TITLE
[BREAK CHANGE] feat: replace with ckb merkle proof and fix tests

### PR DESCRIPTION
**Pull request by magicalne**

### CODE REVIEW

**Comment by jjyr** [comment]
Can you upgrade this PR?

**Comment by magicalne** [comment]
I will push the updates. But tests haven't been fixed yet.

**Comment by jjyr** [comment]
Is this PR ready for merge? can we change the status to `ready for review`?

**Comment by magicalne** [comment]
Sure. Just need to update cargo dependency after https://github.com/nervosnetwork/godwoken/pull/318 is merged.

File: tests/src/script_tests/state_validator/cancel_challenge/tx_execution.rs [hunk 1]
```diff
@@ -58,19 +59,13 @@ fn test_cancel_tx_execute() {
     let finality_blocks = 10;
     let rollup_config = RollupConfig::new_builder()
         .challenge_script_type_hash(Pack::pack(&challenge_script_type_hash))
-        .allowed_eoa_type_hashes(PackVec::pack(allowed_eoa_type_hashes))
```
Comment (jjyr) [inline tests/src/script_tests/state_validator/cancel_challenge/tx_execution.rs:61]: We should use eoa_lock_type_hash and remove `ALWAYS_SUCCESS_CODE_HASH`.

File: tests/src/script_tests/state_validator/cancel_challenge/withdrawal.rs [hunk 2]
```diff
@@ -52,11 +57,13 @@ fn test_cancel_withdrawal() {
     // setup chain
     let mut account_lock_manage = AccountLockManage::default();
     account_lock_manage.register_lock_algorithm(eoa_lock_type_hash.into(), Box::new(AlwaysSuccess));
+    account_lock_manage.register_lock_algorithm(eoa_lock_type_hash.into(), Box::new(AlwaysSuccess));
```
Comment (jjyr) [inline tests/src/script_tests/state_validator/cancel_challenge/withdrawal.rs:60]: Duplicated

**Review by jjyr — CHANGES_REQUESTED** [review]

**Pull request merged**